### PR TITLE
Fix pypi url usage

### LINKED ISSUE #1137: Investigate If Changing Pypi URL in settings is working
Investigate this:

I tried changing my "Pypi URL" in Admin Console to my company's mirror, but when I upload a .whl package in Packages in Admin Console it still tries the typical one. I did click "save" and refresheed the page to see it was there. Do I need to do anything else to get it to use the mirror?

Running Podman on RHEL 9.2, COSMOS 5.14.2. USER/GROUP are both root.

Error:

Stdout:
Could not fetch URL https://pypi.org/simple/pip/: There was a problem confirming the ssl certificate: HTTPSConnectionPool(host='pypi.org', port=443): Max retries exceeded with url: /simple/pip/ (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1006)'))) - skipping

Stderr:
WARNING: The directory '/openc3/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.
ERROR: numpy-1.19.5-cp36-cp36m-manylinux2010_x86_64.whl is not a supported wheel on this platform.

### PR DESCRIPTION
**Pull request by ryanmelt**
closes #1137

### CODE REVIEW

File: openc3/lib/openc3/models/gem_model.rb [hunk 1]
```diff
@@ -36,7 +36,7 @@ module OpenC3
   # and destroy to allow interaction with gem files from the PluginModel and
   # the GemsController.
   class GemModel
-    include Api
+    extend Api
```
Comment (jmthomas) [inline openc3/lib/openc3/models/gem_model.rb:39]: Doesn't extend add as class methods vs include adds as instance methods

File: openc3/lib/openc3/models/python_package_model.rb [hunk 2]
```diff
@@ -71,16 +74,23 @@ def self.install(name_or_path, scope:)
...
+          pypi_url = ENV['PYPI_URL']
+          if pypi_url
+            pypi_url += '/simple'
+          end
+          pypi_url ||= 'https://pypi.org/simple'
+        end
       end
       Logger.info "Installing python package: #{name_or_path}"
-      result = OpenC3::ProcessManager.instance.spawn(["pip", "install", "--user", "-i", pypi_url, package_file_path], "package_install", package_filename, Time.now + 3600.0, scope: scope)
+      result = OpenC3::ProcessManager.instance.spawn(["/openc3/bin/pipinstall", "--user", "--no-warn-script-location", "-i", pypi_url, package_file_path], "package_install", package_filename, Time.now + 3600.0, scope: scope)
```
Comment (jmthomas) [inline openc3/lib/openc3/models/python_package_model.rb:93]: Looks like we were using this in the plugin_model.rb but not here. Is this the real fix?

File: openc3/lib/openc3/models/python_package_model.rb [hunk 3]
```diff
@@ -71,16 +74,23 @@ def self.install(name_or_path, scope:)
...
+        Logger.error("Failed to retrieve pypi_url: #{e.formatted}")
+      ensure
+        if pypi_url.nil?
+          # If Redis isn't running try the ENV, then simply pypi.org/simple
+          pypi_url = ENV['PYPI_URL']
+          if pypi_url
+            pypi_url += '/simple'
+          end
+          pypi_url ||= 'https://pypi.org/simple'
+        end
```
Comment (jmthomas) [inline openc3/lib/openc3/models/python_package_model.rb:90]: Logic is cleaner but basically the same?

**Review by jmthomas — APPROVED** [review]

File: openc3/lib/openc3/models/gem_model.rb (hunk 1 above)
Comment (ryanmelt) [reply openc3/lib/openc3/models/gem_model.rb:39]: Yep.  That was the bug as we were using get_setting in a class method not an instance method.

File: openc3/lib/openc3/models/python_package_model.rb (hunk 2 above)
Comment (ryanmelt) [reply openc3/lib/openc3/models/python_package_model.rb:93]: This adds a retry if the remote install fails, that try to just do local.

File: openc3/lib/openc3/models/python_package_model.rb (hunk 3 above)
Comment (ryanmelt) [reply openc3/lib/openc3/models/python_package_model.rb:90]: Basically the same.

**Pull request merged**